lib/cache/cache.go

```diff
@@ -1771,16 +1853,37 @@ func (c *Cache) fetch(ctx context.Context, confirmedKinds map[resourceKind]types
 // the event will be emitted via the fanout.
 func (c *Cache) processEvent(ctx context.Context, event types.Event) error {
 	resourceKind := resourceKindFromResource(event.Resource)
-	collection, ok := c.collections.byKind[resourceKind]
-	if !ok {
+
+	legacyCollection, legacyFound := c.legacyCacheCollections.byKind[resourceKind]
+	handler, handlerFound := c.collections.byKind[resourceKind]
+
+	switch {
+	case !legacyFound && !handlerFound:
 		c.Logger.WarnContext(ctx, "Skipping unsupported event",
 			"event_kind", event.Resource.GetKind(),
 			"event_sub_kind", event.Resource.GetSubKind(),
 		)
 		return nil
-	}
-	if err := collection.processEvent(ctx, event); err != nil {
-		return trace.Wrap(err)
+	case legacyFound:
+		if err := legacyCollection.processEvent(ctx, event); err != nil {
+			return trace.Wrap(err)
+		}
+	case handlerFound:
+		switch event.Type {
+		case types.OpDelete:
+			if err := handler.onDelete(event.Resource); err != nil {
+				if !trace.IsNotFound(err) {
```

Comment:
`handler` is a dedicated object for receiving changes from our event streams, why are we returning a `*trace.NotFound` error if we don't find the resource we were told to delete by the upstream?

Reply:
`collectionsHandler.onDelete` is new in this PR, we are not forced to keep making the same design mistakes.